source/adapters/hip/program.cpp

```diff
@@ -408,7 +408,10 @@ urProgramGetInfo(ur_program_handle_t hProgram, ur_program_info_t propName,
   case UR_PROGRAM_INFO_BINARIES:
     return ReturnValue(&hProgram->Binary, 1);
   case UR_PROGRAM_INFO_KERNEL_NAMES:
-    return getKernelNames(hProgram);
+    /* TODO: Add implementation for getKernelNames */
+    UR_ASSERT(getKernelNames(hProgram), UR_RESULT_ERROR_UNSUPPORTED_FEATURE);
```

Comment:
I'm aware this was copied from CUDA but I don't think I like using `UR_ASSERT` for checking error codes. This is essentially saying that if `getKernelNames` returns `UR_SUCCESS`, it should in fact return `UR_RESULT_ERROR_UNSUPPORTED_FEATURE`? That's confusing, especially as `getKernelNames` unconditionally returns `UR_RESULT_ERROR_UNSUPPORTED_FEATURE` anyway.

Reply:
Agreed, that code is a bit silly, just updated it for CUDA/HIP to remove the extra function and asserts.